### PR TITLE
Add ntrip client [main branch] [CPP-952]

### PR DESCRIPTION
**Pull request by adrian-kong**
Ports NTRIP client support to the main branch: https://github.com/swift-nav/swift-toolbox/pull/1033

### CODE REVIEW

**Comment by swiftnav-travis** [comment]
Frontend and Release Workflow Started [here](https://github.com/swift-nav/toolbox-backend-ci/actions/runs/5330137573)

**Comment by swiftnav-travis** [comment]
Frontend and Release Workflow Started [here](https://github.com/swift-nav/toolbox-backend-ci/actions/runs/5340274349)

**Comment by swiftnav-travis** [comment]
Frontend and Release Workflow Started [here](https://github.com/swift-nav/toolbox-backend-ci/actions/runs/5340805927)

**Comment by swiftnav-travis** [comment]
Frontend and Release Workflow Started [here](https://github.com/swift-nav/toolbox-backend-ci/actions/runs/5341170358)

**Review by silverjam — APPROVED** [review]

File: resources/SettingsTabComponents/SettingsPane.qml [hunk 1]
```diff
@@ -71,7 +71,7 @@ Rectangle {
             columns: 7
             rows: 12
             rowSpacing: 1
-            height: parent.parent.height + Constants.settingsTab.paneScrollBufferHeight
+            Layout.preferredHeight: parent.height + Constants.settingsTab.paneScrollBufferHeight
```
Comment (pcrumley) [inline resources/SettingsTabComponents/SettingsPane.qml:74]: these fixes look like unrelated bugfixes... just confirming

**Review by pcrumley — APPROVED** [review]
LGTM

File: resources/SettingsTabComponents/SettingsPane.qml (hunk 1 above)
Comment (adrian-kong) [reply resources/SettingsTabComponents/SettingsPane.qml]: ah yea. will revert this height change.

will keep the parents -> parent changes since typo (reference error)

**Comment by swiftnav-travis** [comment]
Frontend and Release Workflow Started [here](https://github.com/swift-nav/toolbox-backend-ci/actions/runs/5383487228)

**Comment by adrian-kong** [comment]
moved to SCD board https://swift-nav.atlassian.net/browse/SCD-4
